frontend_vue/src/components/ui/IncidentDetails.vue

```diff
@@ -1,80 +1,145 @@
-            'Accept-Encoding': 'gzip, deflate', 'Accept-Language':
-            'en-GB,en-US;q=0.9,en;q=0.8,zh-CN;q=0.7,zh;q=0.6,it;q=0.5'}
-          </code>
-        </li>
-      </ul>
-    </section>
+            <template v-if="isNotEmptyObject(val)">
+              <li
+                v-for="(subval, subkey) in val"
+                :key="subkey"
+                class="break-words"
+              >
+                <span class="block text-xs uppercase text-grey-500">{{
+                  subkey
+                }}</span>
+                <span :class="addClassToBoolean(val)">{{ subval }}</span>
+              </li>
+            </template>
+            <template v-else>
+              <li
+                :key="key"
+                :class="addClassToBoolean(val)"
+                class="break-words"
+              >
+                {{ val }}
+              </li>
+            </template>
+          </ul>
+        </template>
+      </section>
+    </div>
   </div>
 </template>
 
 <script setup lang="ts">
-const emit = defineEmits<{
-  (e: 'update:modelValue', value: boolean): void;
+import { computed } from 'vue';
+import type { HitsType } from '@/components/tokens/types.ts';
+import { convertUnixTimeStampToDate } from '@/utils/utils';
+
+const props = defineProps<{
+  hitAlert: HitsType;
 }>();
+
+const emit = defineEmits(['close']);
+
+console.log(props.hitAlert, 'hitsList');
+
+function isNotEmptyObject(
+  val: boolean | string[] | undefined | HitsType
+): boolean {
+  return typeof val === 'object' && Object.keys(val)?.length > 0;
+}
+
+function addClassToBoolean(val: boolean | HitsType | undefined) {
+  if (typeof val === 'boolean') {
+    return val ? 'text-green' : 'text-red';
+  }
+}
+
+function cleanupObject(obj: keyof HitsType) {
+  return Object.entries(obj).reduce((acc, [key, value]) => {
+    if (value !== null && value !== undefined) {
+      //@ts-ignore
+      acc[key as keyof HitsType] =
+        typeof value === 'object' ? cleanupObject(value) : value;
+    }
+    console.log(acc);
+    return acc;
+  }, {} as HitsType);
+}
+
+const formatAlertObject = computed(() => {
+  const geoInfo = cleanupObject(
+    props.hitAlert.geo_info as unknown as keyof HitsType
+  );
+  const requestHeaders = cleanupObject(
+    props.hitAlert.request_headers as unknown as keyof HitsType
+  );
+  const additionalInfo = cleanupObject(
+    props.hitAlert.additional_info as unknown as keyof HitsType
+  );
+  const requestArgs = cleanupObject(
+    props.hitAlert.request_args as unknown as keyof HitsType
+  );
+
+  return {
+    ...(isNotEmptyObject(geoInfo) && { 'Geo Info': geoInfo }),
+    'Is Tor Rely': props.hitAlert.is_tor_relay,
+    ...(isNotEmptyObject(requestHeaders) && {
+      'Request Headers': requestHeaders,
+    }),
+    ...(isNotEmptyObject(additionalInfo) && {
+      'Additional Info': additionalInfo,
+    }),
+    ...(isNotEmptyObject(requestArgs) && {
+      'Request Args': requestArgs,
+    }),
+  };
```

Comment:
this is super hard to read, can we make it clearer that it's returning a dict with so and so keys / values

Reply:
do you mean by adding a comment? or splitting the code somehow?

Reply:
so i'm not sure what this thing is returning at a glance

Reply:
is it just an object with keys including things like `'Is Tor Rely': props.hitAlert.is_tor_relay,`. i'm not sure what this `...(isNotEmptyObject(geoInfo) && { 'Geo Info': geoInfo })` is doing?

Reply:
yeah it's not quite straight-forward, let me try if I can make it look better